### PR TITLE
Update flow to handle branching

### PR DESCRIPTION
**Pull request by reggieb**
- Use hash to define flows
- Add flow unit test
- Pass session to flow so that branching can be based on session data

[Trello ticket](https://trello.com/c/XTaHue3a/443-session-controller-update-flow-to-handle-branching)

### CODE REVIEW

**Comment by reggieb** [comment]
Currently needs rebasing before it can be merged. That will need quite a lot of work - so as development is being frozen on the *session_answers* this branch may need to be left unmerged.

File: app/session_flows/coronavirus_find_support_flow.rb [hunk 1]
```diff
@@ -1,32 +1,85 @@
...
+      self_employed: node_after_self_employed,
+      have_you_been_made_unemployed: node_after_have_you_been_made_unemployed,
+      are_you_off_work_ill: next_group_start,
+      worried_about_work: next_group_start,
+      have_somewhere_to_live: :have_you_been_evicted,
+      have_you_been_evicted: next_group_start,
+      mental_health_worries: :nation,
+      nation: :results,
+    }
   end
```
Comment (gclssvglx) [inline app/session_flows/coronavirus_find_support_flow.rb:46]: Hi @reggieb Nice!  I like this approach, much cleaner.  And all the flow logic is in one place.

File: test/unit/session_flows/coronavirus_find_support_flow_test.rb [hunk 2]
```diff
@@ -0,0 +1,175 @@
+require "test_helper"
+
+class CoronavirusFindSupportFlowTest < ActiveSupport::TestCase
```
Comment (gclssvglx) [inline test/unit/session_flows/coronavirus_find_support_flow_test.rb:3]: Very nice set of tests.  Thanks.

**Review by gclssvglx — APPROVED** [review]
@reggieb Nice one!  More than happy to freeze the dev at this point.

**Pull request merged**